connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Loggers.java

```diff
@@ -151,15 +164,13 @@ public synchronized List<String> setLevel(String namespace, Level level) {
     private synchronized List<org.apache.logging.log4j.Logger> loggers(String namespace) {
         Objects.requireNonNull(namespace, "Logging namespace may not be null");
 
-        if (ROOT_LOGGER_NAME.equalsIgnoreCase(namespace)) {
-            List<org.apache.logging.log4j.Logger> result = currentLoggers();
-            result.add(rootLogger());
-            return result;
+        if (isValidRootLoggerName(namespace)) {
+            return new ArrayList<>(currentLoggers().values());
         }
 
-        List<org.apache.logging.log4j.Logger> result = new ArrayList<>();
-        org.apache.logging.log4j.Logger ancestorLogger = lookupLogger(namespace);
-        List<org.apache.logging.log4j.Logger> currentLoggers = currentLoggers();
+        var result = new ArrayList<org.apache.logging.log4j.Logger>();
+        var ancestorLogger = lookupLogger(namespace);
```

Comment:
maybe we should check all `currentLoggers` and then create the logger for the input namespace if it is nonexistent?